tools/make.ps1

```diff
@@ -11,404 +11,426 @@ $releasePage    = "https://github.com/KoffeinFlummi/armake/releases"
+    }
 }
 
-
-function Remove {
-	$origLocation = Get-Location
-	Set-Location "$projectRoot\.build"
-
-	Switch ($remove) {
-		"all" { Remove-Item "$buildPath" -Recurse -ErrorAction SilentlyContinue }
-		"extras" {
-			$items = $(Get-Extras "Name") -join ","
-			iex "Remove-Item -Path '$buildPath\*' -include $items"
-		}
-		"addons" { Remove-Item "$buildPath\addons\*" }
-		"cache" { Remove-Item "$cachePath" -Recurse -ErrorAction SilentlyContinue }
-	}
-
-	Set-Location $origLocation
+function Remove-Items {
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param()
+
+    $origLocation = Get-Location
+    Set-Location -Path "$projectRoot\.build"
+
+    Switch ($remove) {
+        "all" {
+            if ($PSCmdlet.ShouldProcess("$buildPath", "Remove all items")) {
+                Remove-Item -Path "$buildPath" -Recurse -ErrorAction SilentlyContinue
+            }
+        }
+        "extras" {
+            $items = Get-SupportFiles -type "Name"
+            foreach ($item in $items) {
+                if ($PSCmdlet.ShouldProcess("$buildPath\*", "Remove extras")) {
+                    Remove-Item -Path "$buildPath\*" -Include $item -Force
+                }
+            }
+        }
+        "addons" {
+            if ($PSCmdlet.ShouldProcess("$buildPath\addons\*", "Remove addons")) {
+                Remove-Item -Path "$buildPath\addons\*" -Force
+            }
+        }
+        "cache" {
+            if ($PSCmdlet.ShouldProcess("$cachePath", "Remove cache")) {
+                Remove-Item -Path "$cachePath" -Recurse -ErrorAction SilentlyContinue
+            }
+        }
+    }
+
+    Set-Location -Path $origLocation
 }
 
+function Compare-Version {
+    param(
+        [Parameter(Mandatory=$True)]
+        $version1,
 
-function Compare-VersionNewerThan {
-	param(
-		[Parameter(Mandatory=$True)]
-		$version1,
-
-		[Parameter(Mandatory=$True)]
-		$version2
-	)
+        [Parameter(Mandatory=$True)]
+        $version2
+    )
 
-	$version1 = $version1.Split(".") | % {[int] $_}
-	$version2 = $version2.Split(".") | % {[int] $_}
+    $version1 = $version1.Split(".") | ForEach-Object {[int] $_}
+    $version2 = $version2.Split(".") | ForEach-Object {[int] $_}
 
-	$newer = $False
-	for ($i = 0; $i -lt $version1.Length; $i++) {
-		if ($version1[$i] -gt $version2[$i]) {
-			$newer = $True
-			break
-		}
-	}
+    $newer = $False
+    for ($i = 0; $i -lt $version1.Length; $i++) {
+        if ($version1[$i] -gt $version2[$i]) {
+            $newer = $True
+            break
+        }
+    }
 
-	$newer
+    $newer
 }
 
-
 function Get-InstalledArmakeVersion {
-	if (Test-Path -Path $armake) {
-		$version = & $armake --version
-		$version = $version.Substring(1)
-	} else {
-		$version = "0.0.0"
-	}
-
-	$version
+    if (Test-Path -Path $armake) {
+        $version = & $armake --version
+        $version = $version.Substring(1)
+    } else {
+        $version = "0.0.0"
+    }
+
+    $version
 }
 
-
 function Get-LatestArmakeVersion {
```

Comment:
The original version of this function might work if you replaced used `releases/tag` instead of `releases/download`.
E.g `'href="/KoffeinFlummi/armake/releases/tag/v([\d\.]+)"'`

Reply:
unfortunately not, this is one of the fixes i tried for that issue before giving up, it still fails to find a valid armake download link,

the download link for the latest release is 
https://github.com/KoffeinFlummi/armake/releases/download/v0.6.3/armake_v0.6.3.zip